content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -796,4 +796,90 @@ webhooks:
   ...
 ```
 
+## Best practices and warnings
+
+### Idempotence
+
+An idempotent mutating admission webhook is able to successfully process an object it has already admitted
+and potentially modified. The admission can be applied multiple times without changing the result beyond
+the initial application.
+
+#### Example of idempotent mutating admission webhooks:
+
+1. For a `CREATE` pod request, set the field `.spec.securityContext.runAsNonRoot` of the
+   pod to true, to enforce security best practices.
+
+2. For a `CREATE` pod request, if the field `.spec.containers[].resouces.limits`
+   of a container is not set, set default resource limits.
+
+3. For a `CREATE` pod request, inject a sidecar container with name `foo-sidecar`.
+
+In the cases above, the webhook can be safely reinvoked, or admit an object that already has the fields set.
+
+#### Example of non-idempotent mutating admission webhooks:
+
+1. For a `CREATE` pod request, inject a sidecar container with name `foo-sidecar`
+   suffixed with the current timestamp (e.g. `foo-sidecar-19700101-000000`).
+
+In the case above, reinvoking the webhook can result in the same sidecar being injected multiple times to a pod, each time
+with a different container name. Similarly the webhook can inject duplicated containers if the sidecar already exists in
+a user-provided pod.
+
+### Intercepting all versions of an object
+
+It is recommended that admission webhooks should always intercpet all versions of an object by setting `.webhooks[].matchPolicy`
+to `Equivalent`. It is also recommended that admission webhooks should prefer registering for stable versions of resources.
+Failure to intercept all versions of an object can result in admission policies not being enforced for requests in certain
+versions. See [Matching requests: matchPolicy](#matching-requests-matchpolicy) for examples.
+
+### Availability
+
+It is recommended that admission webhooks should should evaluate as quickly as possible, since they add to API request latency.
```

Comment:
```suggestion
It is recommended that admission webhooks should should evaluate as quickly as possible (typically in milliseconds), since they add to API request latency.
```

Add concrete recommendations here, or at least give an order of magnitude